apis/catalog/v1alpha1/elasticsearch_version_types.go

```diff
@@ -53,15 +53,14 @@ type ElasticsearchVersion struct {
 type ElasticsearchVersionSpec struct {
 	// Version
 	Version string `json:"version" protobuf:"bytes,1,opt,name=version"`
+	// Distribution
+	Distribution ElasticsearchDistro `json:"distribution,omitempty" protobuf:"bytes,2,opt,name=distribution,casttype=ElasticsearchDistro"`
 	// Authentication plugin used by Elasticsearch cluster.
-	AuthPlugin ElasticsearchAuthPlugin `json:"authPlugin" protobuf:"bytes,2,opt,name=authPlugin,casttype=ElasticsearchAuthPlugin"`
+	AuthPlugin ElasticsearchAuthPlugin `json:"authPlugin" protobuf:"bytes,3,opt,name=authPlugin,casttype=ElasticsearchAuthPlugin"`
```

Comment:
With the `Distribution` field, the `AuthPlugin` field won't be needed anymore. Can we make it deprecated?